motorway/intersection.py

```diff
@@ -174,8 +190,47 @@ def receive_messages(self, context=None, output_stream=None, grouper_cls=None):
         set_timeouts_on_socket(receive_sock)
 
         if self.send_control_messages:
-            while not self.controller_sock:
-                time.sleep(1)
+            self._wait_for_controller_sock()
 
         while True:
             self._process(receive_sock, controller_sock=self.controller_sock)
+
+    def report_message_being_processed(self, context=None):
+        socket = context.socket(zmq.REP)
+        self.report_address = socket.bind_to_random_port('tcp://*')
+
+        while True:
+            _ = socket.recv()  # wait for request from WebserverIntersection
+
+            message_being_processed = self.message_being_processed
+            if message_being_processed:
```

Comment:
These two variables are named very similarly (` msgs_being_processed` and `message_being_processed`) and are used almost in the same way.

Maybe
`message_being_processed = self.message_being_processed or []`
and then 
`message_being_processed = self._format_message_being_processed(message_being_processed)`